src/utils/tests/hex-zero-pad.test.ts

```diff
@@ -0,0 +1,42 @@
+import * as ethers from 'ethers';
+import { hexZeroPad } from '../../index';
+
+describe('hex-zero-pad', () => {
+  it('should reject strings passed in which are not hex strings', () => {
+    const values = [
+      '52908400098527886E0F7030069857D2E4169EE7',
+      '8617E340B3D01FA5F11F306F4090FD50E238070D',
+      'de709f2102306220921060314715629080e2fb77',
+      '27b1fdb04752bbc536007a920d24acb045561c26',
+    ];
+    values.forEach((value) => {
+      expect(() => {
+        hexZeroPad(value, 23);
+      }).toThrow(`Value passed in is not a hex string. String: "${value}"`);
+    });
+  });
+  it('should return value passed in when value is longer than that padded string would be', () => {
+    const values = ['0x5aAebAd', '0xfB691', '0xdbF036FB', '0xD1220ab'];
+    values.forEach((value) => {
+      expect(hexZeroPad(value, 2)).toStrictEqual(value);
+    });
+  });
+  it('should match ethers.js and web3.js when padding can be applied', () => {
+    // ether.js chooses to throw an error when the string is longer than the padded string would be
```

Comment:
Oops, I forgot to remove the web3 refernece there, thanks for the catch! I had initially included a web3 test until I realized that web3 does this function differently and wasn't worth comparing to.

Here's how they behave when that situation arises (padding can't be applied because string is already longer than the padded string):
`essential-eth` -> returns the string value without alteration
`ethers` -> throws an `InvalidArgument` error referencing the `value` string passed in
`web3` -> returns the string value without alteration

Choosing to return the string was just from my own judgement of what would be the most valuable to a user. Instead of having to call the function, see if it errs, and then handle the error, just add padding if possible. This is how Node's `padStart` function handles it out of the box as well. Let me know if you have other thoughts though since you've used ethers much more than I have 😛